Credentialer/ViewController.swift

```diff
@@ -8,18 +8,105 @@
+        
+        // If active text field is hidden by keyboard, scroll it so it's visible
+        // Your app might not need or want this behavior.
+        var aRect: CGRect = self.view.frame
+        aRect.size.height -= keyboardSize.height
+        let activeTextFieldRect: CGRect? = activeTextField?.frame
+        let activeTextFieldOrigin: CGPoint? = activeTextFieldRect?.origin
+        if (!CGRectContainsPoint(aRect, activeTextFieldOrigin!)) {
+            scrollView.scrollRectToVisible(activeTextFieldRect!, animated:true)
+        }
+    }
+    
+    func keyboardWillBeHidden(sender: NSNotification) {
+        let contentInsets: UIEdgeInsets = UIEdgeInsetsZero
+        scrollView.contentInset = contentInsets
+        scrollView.scrollIndicatorInsets = contentInsets
+    }
+    
+    func textFieldDidBeginEditing(textField: UITextField) {
+        activeTextField = textField
+        scrollView.scrollEnabled = true
+    }
+    
+    func textFieldDidEndEditing(textField: UITextField) {
+        activeTextField = nil
+        scrollView.scrollEnabled = false
+    }
+    
+    func getActiveTextField() {
+        usernameTextField.addTarget(self, action: Selector("textFieldDidBeginEditing:"), forControlEvents: UIControlEvents.EditingDidBegin)
+        usernameTextField.addTarget(self, action: Selector("textFieldDidEndEditing:"), forControlEvents: UIControlEvents.EditingDidEnd)
+        passwordTextField.addTarget(self, action: Selector("textFieldDidBeginEditing:"), forControlEvents: UIControlEvents.EditingDidBegin)
+        passwordTextField.addTarget(self, action: Selector("textFieldDidEndEditing:"), forControlEvents: UIControlEvents.EditingDidEnd)
+        cityTextField.addTarget(self, action: Selector("textFieldDidBeginEditing:"), forControlEvents: UIControlEvents.EditingDidBegin)
+        cityTextField.addTarget(self, action: Selector("textFieldDidEndEditing:"), forControlEvents: UIControlEvents.EditingDidEnd)
+        zipCodeTextField.addTarget(self, action: Selector("textFieldDidBeginEditing:"), forControlEvents: UIControlEvents.EditingDidBegin)
+        zipCodeTextField.addTarget(self, action: Selector("textFieldDidEndEditing:"), forControlEvents: UIControlEvents.EditingDidEnd)
+        countryTextField.addTarget(self, action: Selector("textFieldDidBeginEditing:"), forControlEvents: UIControlEvents.EditingDidBegin)
+        countryTextField.addTarget(self, action: Selector("textFieldDidEndEditing:"), forControlEvents: UIControlEvents.EditingDidEnd)
+    }
+    
+    func textFieldShouldReturn(textField: UITextField) -> Bool {
+        switch (textField){
+        case self.usernameTextField: self.passwordTextField.becomeFirstResponder()
+        case self.passwordTextField: self.cityTextField.becomeFirstResponder()
+        case self.cityTextField: self.zipCodeTextField.becomeFirstResponder()
+        case self.zipCodeTextField: self.countryTextField.becomeFirstResponder()
+        default:
+            println("")
```

Comment:
You could use `NSAssert` here... checkout this [link] (http://stackoverflow.com/questions/1375786/whats-the-point-of-nsassert-actually)